corc-core/src/main/java/com/hotels/corc/mapred/CorcInputFormat.java

```diff
@@ -189,8 +190,11 @@ static void setReadColumns(Configuration conf, StructTypeInfo actualStructTypeIn
         ids.add(i);
         names.add(actualName);
       }
-      LOG.debug("Set column projection on columns: {} ({})", ids, readNames);
     }
+    if (ids.size() == 0) {
+      throw new IllegalStateException("None of the selected columns were found in the ORC file.");
```

Comment:
Consider "None of the selected columns *was* found" depending on your take on whether none should be paired with a singular verb or not but this is for the grammar nerds so probably not that important here ;)